version: '3.1'
services:
  service:
    build: .
    restart: on-failure
    environment:
      - "MONGO_URI=mongodb://mongo1:27017,mongo2:27018,mongo3:27019/?replicaSet=rs0"
    ports:
      - "6969:6969"
  mongo1:
    hostname: mongo1
    image: mongo:5.0-focal
    ports:
      - 27017:27017
    restart: on-failure
    entrypoint: [ "/usr/bin/mongod", "--bind_ip_all", "--replSet", "rs0" ]
  mongo2:
    hostname: mongo2
    image: mongo:5.0-focal
    ports:
      - 27018:27018
    restart: on-failure
    entrypoint: [ "/usr/bin/mongod", "--bind_ip_all", "--replSet", "rs0", "--port", "27018"]
  mongo3:
    hostname: mongo3
    image: mongo:5.0-focal
    ports:
      - 27019:27019
    restart: on-failure
    entrypoint: [ "/usr/bin/mongod", "--bind_ip_all", "--replSet", "rs0",  "--port", "27019"]
  mongosetup:
    image: mongo:5.0-focal
    depends_on:
      - mongo1
      - mongo2
      - mongo3
    volumes:
      - .:/scripts
    restart: "no"
    entrypoint: [ "bash", "/scripts/mongo_setup.sh"]
  proxy:
    image: envoyproxy/envoy-dev:436946bb9df0acda0e4e709592205f0d199dfb79
    restart: on-failure
    ports:
      - "8080:8080"
    volumes:
      - ./envoy.yaml:/etc/envoy/envoy.yaml
    extra_hosts:
      host.docker.internal: host-gateway
  rabbitmq:
    image: docker.io/bitnami/rabbitmq:3.8
    ports:
      - '4369:4369'
      - '5672:5672'
      - '25672:25672'
      - '15672:15672'
    environment:
      - RABBITMQ_SECURE_PASSWORD=yes
    volumes:
    - ./rabbitmq.conf:/bitnami/rabbitmq/conf/custom.conf:ro
